homework3/task1.c

```diff
@@ -0,0 +1,132 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <stdbool.h>
+
+#define arraySize 30
+
+void swap(int *firstElement, int *secondElement)
+{
+    int transitElement = *firstElement;
+    *firstElement = *secondElement;
+    *secondElement = transitElement;
+}
+
+
+int partition(int *array, int lowIndex, int highIndex)
+{
+    int pivot = array[lowIndex];
+    int pivotIndex = lowIndex;
+
+
+    int lastIndexOfPivot = 0;
+
+    for (int i = lowIndex + 1; i < highIndex; ++i)
+    {
+        if (array[i] <= pivot)
```

Comment:
```suggestion
        if (array[i] < pivot)
```
При таком знаке мы получим корректное разделение, а перестановок элементов станет меньше